### PR TITLE
Fix large file uploads through cockpit.spawn()

### LINKED ISSUE #19323: AttributeError: 'SubprocessStreamChannel' object has no attribute '_last_ping'
I was testing cockpit-navigator and file uploading. On Fedora 38,

 - `sudo dnf install cockpit-navigator`
 - Upload a large file (I tried a 216 MB video file) to the cockpit VM with navigator

This results in a bridge crash, and thousands of these journal messages:
```
asyncio-ERROR: Exception in callback _Transport._write_ready()
handle: <Handle _Transport._write_ready()>
Traceback (most recent call last):
  File "/usr/lib64/python3.11/asyncio/events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
  File "/usr/lib/python3.11/site-packages/cockpit/transports.py", line 189, in _write_ready
    self._remove_write_queue()
  File "/usr/lib/python3.11/site-packages/cockpit/transports.py", line 197, in _remove_write_queue
    self._protocol.resume_writing()
  File "/usr/lib/python3.11/site-packages/cockpit/channel.py", line 405, in resume_writing
    if self._last_ping is not None:
       ^^^^^^^^^^^^^^^
AttributeError: 'SubprocessStreamChannel' object has no attribute '_last_ping'
```

Also tracked in RHEL: https://issues.redhat.com/browse/RHEL-5341

### PR DESCRIPTION
**Pull request by martinpitt**
This fixes a bridge crash

    'SubprocessStreamChannel' object has no attribute '_last_ping'

when resume_writing() is called before the initial ping arrived.

Fixes #19323

----

That fixes the crash, but it's only half of the story. The new test case still fails with a too short file size:

```
Traceback (most recent call last):
  File "/var/home/martin/upstream/cockpit/file-upload/test/verify/check-connection", line 831, in testFlowControlUpload
    self.assertEqual(m.execute("stat -c %s /tmp/spawninput").strip(), 65536000)
AssertionError: '52494336' != 65536000
```

Then again, with the C bridge (rhel-8-9) it's even worse: it fails sometimes with
```
  File "/var/home/martin/upstream/cockpit/file-upload/test/verify/check-connection", line 827, in testFlowControlUpload
    b.wait_text_not("#speed", "")
```
as it finishes way too fast, and sometimes with
```
Traceback (most recent call last):
  File "/var/home/martin/upstream/cockpit/file-upload/test/verify/check-connection", line 831, in testFlowControlUpload
    self.assertEqual(m.execute("stat -c %s /tmp/spawninput").strip(), 65536000)
AssertionError: '65536' != 65536000
```
again because it finishes way too fast, and only writes the first block.

Let's make this work properly, so that navigator etc. works, and we generally have a way to support large file uploads in Cockpit.

### CODE REVIEW

**Comment by martinpitt** [comment]
@allisonkarlitskaya and/or @mvollmer could you please have a look while I'm away?

**Comment by allisonkarlitskaya** [comment]
This is going to need rebasing to put the fix before the addition of the integration test (and also to merge the fixup).

File: pkg/playground/speed.js [hunk 1]
```diff
@@ -226,6 +226,31 @@ function spawn() {
...
+        channel.input(chunk, true);
+        total += chunk.length;
+    }
+    channel.input(); // send EOF
+    channel
+            .then(() => {
+                el_result.textContent = "success";
+            })
+            .catch(ex => {
+                el_result.textContent = `failed with exit code ${ex.exit_status}: ${ex.message}`;
```
Comment (cockpituous) [inline pkg/playground/speed.js:249]: These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19325-20230921-094750-868fc656-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Comment by allisonkarlitskaya** [comment]
Cool fail on RHEL8.  I wonder what's going on there...

**Comment by allisonkarlitskaya** [comment]
> Cool fail on RHEL8. I wonder what's going on there...

Oh, of course.  That's the C bridge failing.

File: pkg/playground/speed.js (hunk 1 above)
Comment (cockpituous) [inline pkg/playground/speed.js:249]: These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19325-20230921-094750-868fc656-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Comment by martinpitt** [comment]
@allisonkarlitskaya yep, we'll have to skip the test there; see comment above, it fails even worse on the C bridge. Many thanks for fixing!

**Comment by martinpitt** [comment]
@allisonkarlitskaya Cheers! I cleaned this up, should go green now.

File: pkg/playground/speed.js (hunk 1 above)
Comment (cockpituous) [inline pkg/playground/speed.js:249]: These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19325-20230921-183914-ded91df0-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Review by allisonkarlitskaya — APPROVED** [review]
:hand: ^5

**Pull request merged**